challenges/01 Goldilocks' Bear Necessities/app.js

```diff
@@ -1 +1,77 @@
-// Do awesome stuff...
+class GoldilocksHelper {
+
+    /**
+     * Constuctor
+     * @param  string file Filepath
+     */
+    constructor(file) {
+        this.readFile(file, (err, content) => {
+            if (err)
+                throw err
+            else {
+                this.parseValues(content)
+                this.calculateSuitableChairs()
+                this.output()
+            }
+        })
+    }
+
+    /**
+     * Read File as string
+     * @param  string file Filepath
+     * @param  callback (err,content) Callback
+, after fileread completed
+     */
+    readFile(file, callback) {
+        let fs = require('fs') //require fs library
+        fs.readFile(file, 'utf8', callback) //read file with nodejs
+    }
+
+    /**
+     * Parse values from content string
+     * @param  string content Content string returned from fs.readFile
+     *
+     */
+    parseValues(content) {
+        let lines = content.split('\n') //split content string to array line by line
+        this.weight = parseInt(lines[0].split(' ')[0]) //get weight of Goldilocks from first line left value
+        this.maxTemperature = parseInt(lines[0].split(' ')[1]) //get maximum temperature that Goldilocks can tolerate from first line right value
+        let lineArray
+        this.chairs = []
+        for (let i = 1; i < lines.length; i++) { //iterate through all lines from index = 1
+            lineArray = lines[i].split(' ') //split left and right values
+            this.chairs.push({ weight: parseInt(lineArray[0]), temperature: parseInt(lineArray[1]) }) //parse int from string values and then push them as object to array
+        }
+    }
+
+    /**
+     * Calculates suitable chairs from all value pairs in this.chairs
+     */
+    calculateSuitableChairs() {
+        this.suitableChairs = []
+        this.chairs.forEach((chair, index) => {
+            if (chair.weight >= this.weight && chair.temperature <= this.maxTemperature) // if chair maximum weight is greater than Goldilocks and temperature less than Goldilocks
+                this.suitableChairs.push(index + 1) //add this chair index to suitable chairs array with index +1, because there are no Goldilocks parameters as first arrayelement
```

Comment:
Der erste Kommentar ('if chair maximum weight...') spiegelt nur den Code wieder.
Der zweite Kommentar ('add this chair...') ist schon besser, er beschreibt _warum_ es _index + 1_ ist.